packages/integration-sdk-core/src/types/step.ts

```diff
@@ -93,6 +93,10 @@ export type IntegrationStepResult = Omit<
    * Entity or relationship types encountered during step execution.
    */
   encounteredTypes: string[];
+
+  startTime?: number;
+  endTime?: number;
+  duration?: number;
```

Comment:
```suggestion
  /**
   * Start time of the step in milliseconds since the Unix epoch.
   */
  startTime?: number;

  /**
   * End time of the step in milliseconds since the Unix epoch.
   */
  endTime?: number;

  /**
   * Duration of the step in milliseconds, calculated as endTime - startTime.
   */
  duration?: number;
```